database/database.go

```diff
@@ -132,21 +144,28 @@ func (db *Database) GetFileContent(id int) (obj.File, error) {
 		SELECT filename, data, owner
 		FROM files
 		WHERE id=?`, id)
-	if rows != nil {
-		defer rows.Close()
-	}
 	if err != nil {
 		return f, err
```

Comment:
I really don't understand how this close() stuff works but do you need a `rows.Close()` before this return on line 148?

Reply:
Nope, not here - if there is an error, then rows will be nil and can't be closed. I did another glance over to make sure that I haven't missed any others and it seems good.